src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyImpl.java

```diff
@@ -167,9 +183,26 @@ public void process() throws TransportException {
                         final ProxyHandler.ProxyResponseResult responseResult = proxyHandler
                                 .validateProxyResponse(inputBuffer);
                         inputBuffer.compact();
-
+                        inputBuffer.clear();
                         if (responseResult.getIsSuccess()) {
                             proxyState = ProxyState.PN_PROXY_CONNECTED;
+                        } else if (responseResult.getError() != null &&
+                                responseResult.getError().contains(PROXY_AUTH_DIGEST)) {
+                            proxyState = ProxyState.PN_PROXY_NOT_STARTED;
+                            final Scanner responseScanner = new Scanner(responseResult.getError());
+                            final Map<String, String> challengeQuestionValues = new HashMap<String, String>();
+                            while (responseScanner.hasNextLine()) {

```

Comment:
Think about the scope of this class as -a place holder for the state transistions and buffers. Please refactor processing the challenge headers into a different files. 

I would Create an interface something like:
```
interface ProxyChallengeProcessor {
 string getHeader();
}
```

implement a basic auth processor and digest auth processor. Select processor based on the Challege response.

Reply:
added ProxyChallengeProcessor  and ProxyChallengeProcessorImpl classes